src/Forms/VisualSync.cs

```diff
@@ -752,27 +756,25 @@ private void timerProgressBarRefresh_Tick(object sender, EventArgs e)
             timerProgressBarRefresh.Start();
         }
 
-        private void SetSyncFactorLabel()
+        private void SetSyncFactorLabel(double videoPlayerCurrentStartPos, double videoPlayerCurrentEndPos)
         {
-            _adjustInfo = string.Empty;
             if (string.IsNullOrWhiteSpace(VideoFileName))
                 return;
 
-            double startPos = MediaPlayerStart.CurrentPosition;
-            double endPos = MediaPlayerEnd.CurrentPosition;
-            if (endPos > startPos)
+            _adjustInfo = string.Empty;
+            if (videoPlayerCurrentEndPos > videoPlayerCurrentStartPos)
             {
                 double subStart = _paragraphs[comboBoxStartTexts.SelectedIndex].StartTime.TotalMilliseconds / TimeCode.BaseUnit;
                 double subEnd = _paragraphs[comboBoxEndTexts.SelectedIndex].StartTime.TotalMilliseconds / TimeCode.BaseUnit;
 
                 double subDiff = subEnd - subStart;
-                double realDiff = endPos - startPos;
+                double realDiff = videoPlayerCurrentEndPos - videoPlayerCurrentStartPos;
 
                 // speed factor
                 double factor = realDiff / subDiff;
 
                 // adjust to starting position
-                double adjust = startPos - subStart * factor;
+                double adjust = videoPlayerCurrentStartPos - subStart * factor;

```

Comment:
Was the intention to multiply `subStart * factor`   1st and then subtract? @niksedk